packages/server/lib/controllers/sync.controller.ts

```diff
@@ -118,14 +118,42 @@ class SyncController {
                 return;
             }
 
-            const syncs = await getSyncs(connection, orchestrator);
+            const rawSyncs = await getSyncs(connection, orchestrator);
+
+            if (!connection.id) {
+                res.send(rawSyncs);
+                return;
+            }
+
+            const syncs = await this.addObjectCount(rawSyncs, connection.id);
 
             res.send(syncs);
         } catch (e) {
             next(e);
         }
     }
 
+    private async addObjectCount(syncs: (Sync & { models: string[] })[], connectionId: number) {
+        const models = syncs.map((sync) => sync.models).flat();
+        const objectCountByModel: Record<string, number> = {};
+        for (const model of models) {
+            if (!(model in objectCountByModel)) {
+                const count = await recordsService.getRecordsCount({ connectionId, model });
```

Comment:
Note: Some connections/models might have millions of records. Counting records inline could take several seconds each time and make the call to the endpoint quite slow.

Reply:
Per our discussion will be investigating an alternative approach - store the per-(connection_id, model) count somewhere in the database, and update it at the end of each sync task